### ISSUE TITLE
DoesNotReturn analysis complains about 'return' statements within local functions

### ISSUE THREAD

**Issue by RikkiGibson** [dotnet/roslyn#45795]
**Version Used**: 7b4bfaaee

**Steps to Reproduce**:

```cs
using System.Diagnostics.CodeAnalysis;
#nullable enable
public class C {
    [DoesNotReturn]
    public void M() {
        _ = local1();
        throw null!;
        
        int local1() { return 1; }
    }
}
```

**Expected Behavior**: No warnings

**Actual Behavior**: Warning on 'return 1' within local function
